### ISSUE TITLE
Connection refused on localhost - only with Docker

### ISSUE THREAD

**Issue by Guttz** [HTTP-APIs/hydra-python-agent#104]
### I'm submitting a 
- [x] bug report.
- [ ] feature request.

### Current Behaviour:

When trying to connect to a localhost hydrus server using docker, the agent returns Connection refused as the image:

![image](https://user-images.githubusercontent.com/11739405/58632167-25836300-82e5-11e9-9a88-85a2d90e7a3b.png)

However, it works for http://35.224.198.158:8080/api. At the same time, running python querying_mechanism.py directly works perfectly for both remote and local server.
 
### Expected Behaviour:

Running via docker should enable to connect to local hydrus servers. 

### Do you want to work on this issue?

Yes, but appreciate any suggestions since I'm not really sure the reason for it

**Comment by Guttz** [comment]
The error message received seems to be somehow related to docker not being able to connect to Redis(but not sure)

**Comment by xadahiya** [comment]
Which OS are you using?

**Comment by Guttz** [comment]
I'm on ubuntu

**Comment by motormouthdoda** [comment]
> The error message received seems to be somehow related to docker not being able to connect to Redis(but not sure)

If that was the case, Docker wouldn't have connected to [http://35.224.198.158:8080/api](http://35.224.198.158:8080/api) as well. The problem seems to be of correct port mapping. Kindly take a look here [https://stackoverflow.com/questions/24319662/from-inside-of-a-docker-container-how-do-i-connect-to-the-localhost-of-the-mach](https://stackoverflow.com/questions/24319662/from-inside-of-a-docker-container-how-do-i-connect-to-the-localhost-of-the-mach)

**Comment by Guttz** [comment]
### Quick fix:

Linux:
Run sudo ip addr show docker0
Copy the default docker bridge that it creates, by default it's http://172.17.0.1
Windows/Mac:
use http://host.docker.internal instead of localhost



Seems like eventually using host.docker.internal should work in all OS, but not working yet. There are

**Comment by Guttz** [comment]
### Quick fix:

**Linux:**
`sudo ip addr show docker0`

Copy the default docker bridge link that is created, by default it's http://172.17.0.1

**Windows/Mac:**
Use http://host.docker.internal instead of http://localhost

If you use Windows/Mac and can test this solution and give a thumbs up would be helpful so I can add in the docs. Also, seems like soon host.docker.internal should work in all OS. 

[Follow the post mentioned by @motormouthdoda for more answers. ](https://stackoverflow.com/questions/24319662/from-inside-of-a-docker-container-how-do-i-connect-to-the-localhost-of-the-mach)

**Comment by Mec-iS** [comment]
@Guttz can we put this configuration bit in docker-compose or in a bash script?

**Comment by Guttz** [comment]
I tried it for a while but didn't find something for all OS. I will install a VM to test some other compatibility matters with the agent and then I come back here and take a look again into this.

**Comment by Mec-iS** [comment]
> Quick fix:
Linux:
sudo ip addr show docker0

> Copy the default docker bridge link that is created, by default it's http://172.17.0.1

> Windows/Mac:
Use http://host.docker.internal instead of http://localhost

>If you use Windows/Mac and can test this solution and give a thumbs up would be helpful so I can add in the docs. Also, seems like soon host.docker.internal should work in all OS.

>Follow the post mentioned by @motormouthdoda for more answers.

I suppose we can just add this part to a "Troubleshooting" section in the README

**Comment by Guttz** [comment]
Yes, at least for now, I'll add this there while we can come with a better solution.

There's a PR which would make http://host.docker.internal a universal solution but it doesn't seem like a priority right now: https://github.com/docker/libnetwork/pull/2348

**Comment by Blakelist7** [comment]
Hi, @Guttz I am new to the community, could you help me getting this issue resolved. Also new to open source :)
